packages/wonder-blocks-tabs/src/components/tabs.tsx

```diff
@@ -86,8 +96,90 @@ export const Tabs = React.forwardRef(function Tabs(
         onTabSelected,
         "aria-label": ariaLabel,
         "aria-labelledby": ariaLabelledby,
+        activationMode = "manual",
     } = props;
 
+    const focusedTabId = React.useRef(selectedTabId);
+
+    React.useEffect(() => {
+        focusedTabId.current = selectedTabId;
+    }, [selectedTabId]);
+
+    const selectTab = React.useCallback(
+        (tabId: string) => {
+            if (tabId !== selectedTabId) {
+                // Select the tab only if it's not already selected
+                onTabSelected(tabId);
+            }
+        },
+        [onTabSelected, selectedTabId],
+    );
+
+    const handleKeyInteraction = React.useCallback(
+        (tabId: string) => {
+            // Move focus to the tab
+            const tabElement = document.getElementById(getTabId(tabId));
```

Comment:
I'm trying to think through a way of getting the tabElement without relying on DOM methods. Maybe create an array of refs and filter based on a matching id? The DOM method might just be easier and more performant, though...